### ISSUE TITLE
Database error 22P02: invalid input syntax for integer: "0.0"

### ISSUE THREAD

**Issue by aroekene** [dimitri/pgloader#764]
I'm trying to load data from a MS SQL database onto a PostgreSQL database.
Some columns in MS SQL that are datatype float corresponds with datatype int in the newly created PostgreSQL database (Vendor specific)
Any way to overcome this? The float columns in MS SQL never use the decimal digit (1.0, 2.0, 3.0 etc) and the new database will only use the first digit (1, 2, 3 etc)

When trying to run it now we get the error message Database error 22P02: invalid input syntax for integer: "0.0" and data will not load.
I've tried with some different cast rules and transform functions, but none seem to be working.

```
[root@server pgloader]# ./build/bin/pgloader --version
pgloader version "3.4.a7fd776"
compiled with SBCL 1.4.0-1.el7
```

Running on CentOS Linux release 7.4.1708 (Core)

**Comment by dimitri** [comment]
Can you try with using the transformation function `truncate`, which is a standard Common Lisp function (see http://www.lispworks.com/documentation/HyperSpec/Body/f_floorc.htm#truncate). My guess is that it should work, and if not, I'll have to have a closer look at your problem, but I won't be able to reproduce it.

**Comment by aroekene** [comment]
Thank you for the quick reply. I tried using the truncate function (perhaps i used it wrongly) 
`cast column column_name to int using truncate`

This gives an error
```
2018-03-17T18:57:14.307000+01:00 ERROR The value
  0
is not of type
  STRING
when binding STRING
```

Since this is a non crucial column it seems like I can work around it using the following cast
`cast column column_name to int using ensure-parse-integer`
This just gives me a null value at the destination database, but it doesn't matter in this case.


Thanks again for pointing me in the right direction.

**Comment by aroekene** [comment]
Just wanted to post a update so that others out there may benefit from my solution here.

Since this is a one time transfer of data I found another solution for this.
Since pgloader actually creates tables in the scheme you want i just did a clean transfer of all objects from MS SQL to PostgreSQL. 
MS SQL stores data mainly in its "dbo" scheme and we use the public scheme on Postgres. The tables we transfered were created in the dbo scheme. From there we were able to do a simple insert into:

`insert into public.table select * from dbo.table`
